erpnext/buying/report/item_wise_purchase_history/item_wise_purchase_history.py

```diff
@@ -0,0 +1,278 @@
+	item_details = get_item_details()
+	purchase_order_records = get_purchase_order_details(company_list, filters)
+
+	for record in purchase_order_records:
+		supplier_record = supplier_details.get(record.supplier)
+		item_record = item_details.get(record.item_code)
+		row = {
+			"item_code": record.get("item_code"),
+			"item_name": item_record.get("item_name"),
+			"item_group": item_record.get("item_group"),
+			"description": record.get("description"),
+			"quantity": record.get("qty"),
+			"uom": record.get("uom"),
+			"rate": record.get("base_rate"),
+			"amount": record.get("base_amount"),
+			"purchase_order": record.get("name"),
+			"transaction_date": record.get("transaction_date"),
+			"supplier": record.get("supplier"),
+			"supplier_name": supplier_record.get("supplier_name"),
+			"supplier_group": supplier_record.get("supplier_group"),
+			"project": record.get("project"),
+			"received_qty": flt(record.get("received_qty")),
+			"billed_amt": flt(record.get("billed_amt")),
+			"company": record.get("company"),
+		}
+		row["currency"] = frappe.get_cached_value("Company", row["company"], "default_currency")
+		data.append(row)
+
+	return data
+
+
+def get_supplier_details():
+	details = frappe.get_all("Supplier", fields=["name", "supplier_name", "supplier_group"])
+	supplier_details = {}
+	for d in details:
+		supplier_details.setdefault(
+			d.name,
+			frappe._dict({"supplier_name": d.supplier_name, "supplier_group": d.supplier_group}),
+		)
+	return supplier_details
+
+
+def get_item_details():
+	details = frappe.db.get_all("Item", fields=["name", "item_name", "item_group"])
+	item_details = {}
+	for d in details:
+		item_details.setdefault(d.name, frappe._dict({"item_name": d.item_name, "item_group": d.item_group}))
+	return item_details
+
+
+def get_purchase_order_details(company_list, filters):
+	db_po = frappe.qb.DocType("Purchase Order")
+	db_po_item = frappe.qb.DocType("Purchase Order Item")
+
+	query = (
+		frappe.qb.from_(db_po)
+		.inner_join(db_po_item)
+		.on(db_po_item.parent == db_po.name)
+		.select(
+			db_po.name,
+			db_po.supplier,
+			db_po.transaction_date,
+			db_po.project,
+			db_po.company,
+			db_po_item.item_code,
+			db_po_item.description,
+			db_po_item.qty,
+			db_po_item.uom,
+			db_po_item.base_rate,
+			db_po_item.base_amount,
+			db_po_item.received_qty,
+			(db_po_item.billed_amt * db_po.conversion_rate).as_("billed_amt"),
+		)
+		.where(db_po.docstatus == 1)
+		.where(db_po.company.isin(tuple(company_list)))
+	)
+
+	if filters.get("item_group"):
+		query = query.where(db_po_item.item_group == filters.item_group)
+
+	if filters.get("from_date"):
+		query = query.where(db_po.transaction_date >= filters.from_date)
+
+	if filters.get("to_date"):
+		query = query.where(db_po.transaction_date <= filters.to_date)
+
+	if filters.get("item_code"):
+		query = query.where(db_po_item.item_code == filters.item_code)
+
+	if filters.get("supplier"):
+		query = query.where(db_po.supplier == filters.supplier)
```

Comment:
```suggestion
	for field in ("item_code", "item_group", "supplier"):
		if filters.get(field):
			query = query.where(db_po_item[field] == filters[field])

	if filters.get("from_date"):
		query = query.where(db_po.transaction_date >= filters.from_date)

	if filters.get("to_date"):
		query = query.where(db_po.transaction_date <= filters.to_date)
```

Reply:
@s-aga-r, changes done but

> for field in ("item_code", "item_group", "supplier"):

supplier is not a part of the purchase invoice item so we don't add it. we add it separately.